frontend/src/presentation/components/QuestionList.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react";
+import { List, Spin, Alert } from "antd";
+import { Question } from "../../domain/entities/Question";
+import { QuestionCard } from "./QuestionCard";
+import styles from "./QuestionList.module.css";
+import { SearchBar } from "./SearchBar";
+
+interface QuestionListProps {
+	questions: Question[];
+	selectedQuestion: Question | null;
+	onSelectQuestion: (questionId: string) => void;
+	isNarrow: boolean;
+	isLoading: boolean;
+	error: string | null;
+}
+
+export const QuestionList: React.FC<QuestionListProps> = ({
+	questions,
+	selectedQuestion,
+	onSelectQuestion,
+	isNarrow,
+	isLoading,
+	error,
+}) => {
+	const [filteredQuestions, setFilteredQuestions] =
+		useState<Question[]>(questions);
+	const [searchTerm, setSearchTerm] = useState("");
+	const [filters, setFilters] = useState({
+		difficulty: "",
+		category: "",
+		status: "",
+	});
+
+	useEffect(() => {
```

Comment:
I dont this this useEffect is needed, but correct me if Im wrong. As such, we dont need the `filteredQuestions` state too. Can just do:
`const filteredQuestions = questions.filter(...)`

See: https://react.dev/learn/you-might-not-need-an-effect

Reply:
You are right, thanks for pointing this out